File: backend/EventManagement/Models/Event.cs
namespace EventManagement.Models;

public class Event
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Location { get; set; } = string.Empty;
    public DateTime Date { get; set; }
    public TimeSpan StartTime { get; set; }
}

public class EventRegistration
{
    public int Id { get; set; }
    public int EventId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Pronouns { get; set; } = string.Empty;
    public bool OptInForCommunication { get; set; }
    public DateTime RegistrationDate { get; set; }
}

File: frontend/event-management/src/types/Event.ts
export interface Event {
  id: number
  name: string
  location: string
  date: string
  startTime: string
}

export interface EventRegistration {
  id?: number
  eventId: number
  name: string
  email: string
  pronouns: string
  optInForCommunication: boolean
  registrationDate?: string
}

export interface EventFilter {
  date?: string
  location?: string
}